spec/support/examples_group/shared_examples_for_vm_or_template_is_available.rb

```diff
@@ -1,30 +1,51 @@
 shared_examples_for "Vm operation is available when not powered on" do
   it "when powered on" do
     vm.update_attributes(:raw_power_state => power_state_on)
-    expect(vm.is_available?(state)).to be_falsey
+    if vm.respond_to?("supports_#{state}?")
+      expect(vm.public_send("supports_#{state}?")).to be_falsey
+    else
+      expect(vm.is_available?(state)).to be_falsey
+    end
```

Comment:
Does the test need this logic here?  That is, since this test is specifically for power state, I would think it would only have the current implementation (as opposed to coding for both)